modules/protocol/src/vector.ts

```diff
@@ -39,6 +46,9 @@ export class Vector implements IVectorProtocol {
   // Hold the serialized queue for each channel
   private queues: Map<string, SerializedQueue<SelfUpdateResult, OtherUpdateResult>> = new Map();
 
+  // Hold the merkle tree for each channel
+  private trees: Map<string, merkle.Tree> = new Map();
```

Comment:
How much memory will this realistically consume? It could make sense to prune entries after not being used for a while or based on a maximum number of trees to keep around. A library I can recommend is https://www.npmjs.com/package/@thi.ng/cache, which provides both an LRU cache as well as a TLRU cache.

Reply:
Hmm good question, let me check out that library. Thanks for the tip!

Reply:
This will consume approximately 64 bytes per transfer (lower bound, not counting sizing Vec buffer). So, if the router has a million transfers this would be a "whopping" 64 megabytes. I think this will be ok and that adding something here like an LRU cache isn't worth the complexity/effort/performance. Most LRU caches that I've seen in the wild just make performance worse.